src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,53 @@
+package duke.parser;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.exception.UnknownCommandException;
+
+public class Parser {
+
+    /**
+     * Takes in a user input and returns a duke.command.Command object.
+     * @param userInput
+     * @return duke.command.Command
+     */
+    public static Command parseUserInput(String userInput) throws UnknownCommandException {
+        String[] commandAndArgs = getCommandAndArgs(userInput);
+        String commandType = commandAndArgs[0];
+        String commandArgs = commandAndArgs[1];
+
+        switch (commandType) {

```

Comment:
I like that you've used switch structure instead of nested if structure. You have avoided arrowhead!